besu/src/main/java/org/hyperledger/besu/controller/QbftBesuControllerBuilder.java

```diff
@@ -265,11 +265,44 @@ protected ProtocolSchedule createProtocolSchedule() {
   protected void validateContext(final ProtocolContext context) {
     final BlockHeader genesisBlockHeader = context.getBlockchain().getGenesisBlock().getHeader();
 
-    if (bftBlockInterface().get().validatorsInBlock(genesisBlockHeader).isEmpty()) {
+    if (usingValidatorContractModeButSignersExistIn(genesisBlockHeader)) {
+      LOG.warn(
+          "Using validator contract mode but genesis block contains signers - the genesis block signers will not be used.");
+    }
+
+    if (usingValidatorBlockHeaderModeButNoSignersIn(genesisBlockHeader)) {
       LOG.warn("Genesis block contains no signers - chain will not progress.");
     }
   }
 
+  private boolean usingValidatorContractModeButSignersExistIn(
+      final BlockHeader genesisBlockHeader) {
+    return genesisConfig
+            .getConfigOptions()
+            .getQbftConfigOptions()
+            .getValidatorContractAddress()
+            .isPresent()
+        && signersExistIn(genesisBlockHeader);
+  }
+
+  private boolean usingValidatorBlockHeaderModeButNoSignersIn(
+      final BlockHeader genesisBlockHeader) {
+    return genesisConfig
+            .getConfigOptions()
+            .getQbftConfigOptions()
+            .getValidatorContractAddress()
+            .isEmpty()
+        && noSignersExistIn(genesisBlockHeader);
+  }
+
+  private boolean signersExistIn(final BlockHeader genesisBlockHeader) {
```

Comment:
nit: not convinced this method is adding a lot of value. For the one place this is used I think noSignersExistIn(genesisBlockHeader) is easier to read.

Reply:
In order to keep the positively worded method, I have changed `signersExistIn` and removed noSignersExistIn (using negation instead)